### PR TITLE
CircleCI job for e2e test

### PR DESCRIPTION
**Pull request by john-a-joyce**
Initial config file changes to spawn
a CircleCI e2e test.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,7 +9,87 @@ defaults: &defaults
...
+            - TEST_ENV: minikube-none
+            - GOPATH: /go
+    steps:
+      - <<: *initWorkingDir
+      - checkout
+      - attach_workspace:
+          at:  /go
+      - <<: *markJobStartsOnGCS
+      - run: make sync
+      - run: bin/testEnvRootMinikube.sh start
```
Comment (tiswanso) [inline .circleci/config.yml]: We can make these jobs pull istio/istio from this job definition and invoke the bin/testEnvRootMinikube.sh from there.  IMO, we shouldn't copy it into this repo.

Comment (john-a-joyce) [reply .circleci/config.yml]: Yeah - the job needs to pull istio/istio but my thinking we might have jobs that want to create a minikube environment without istio/istio being pulled.   I guess we can include that when it is needed.   For now I want to get some type of trigger in so it was easier to iterate on a couple different vectors.

File: .circleci/config.yml [hunk 2]
```diff
@@ -9,7 +9,87 @@ defaults: &defaults
...
+      - <<: *markJobFinishesOnGCS
+      - store_artifacts:
+          path: /home/circleci/logs
+      - store_artifacts:
+          path: /tmp
+      - store_artifacts:
+          path: /var/lib/localkube/
+      - store_test_results:
+          path: /go/out/tests
+
```
Comment (tiswanso) [inline .circleci/config.yml]: need to reference this job from the workflows section to get it to run.  The "all" list runs on PRs.

**Comment by tiswanso** [comment]
Closing this as John is on PTO and it conflicts with the install-cni circleci job PR #32 that merged.  All of this change was absorbed into PR #33.

**Pull request closed without merging**